test/pylib/minio_server.py

```diff
@@ -50,8 +50,8 @@ def __init__(self, tempdir_base, address, logger):
         self.default_user = 'minioadmin'
         self.default_pass = 'minioadmin'
         self.bucket_name = 'testbucket'
-        self.access_key = ''.join(random.choice(string.hexdigits) for i in range(16))
-        self.secret_key = ''.join(random.choice(string.hexdigits) for i in range(32))
+        self.access_key = os.environ.get(self.ENV_ACCESS_KEY, ''.join(random.choice(string.hexdigits) for i in range(16)))
+        self.secret_key = os.environ.get(self.ENV_SECRET_KEY, ''.join(random.choice(string.hexdigits) for i in range(32)))
```

Comment:
These values will be written into some temp file in test.py's tempdir , this file is not guaranteed to have any good permissions, nether it's guaranteed to be safely removed after test finishes. Is that OK?

Reply:
Very good question. For CI, probably yes. For a general user? Maybe not...

Reply:
Same for both - we don't want to write credentials to a local file - if we can avoid it by removing the refs from the config and use env vars it's preferred, AFAIK minio can use env vars for authentication

Reply:
The variable names that are used by minio_server.py match standard aws ones "by coincidence". They used to have different names, but then someone suggested to use standard names. We can rename them back not to overwrite aws ones, but then my question is -- what code currently suffers from overwritten (by test.py) standard envs?

Reply:
> Or at least the mechanism should allow for said credentials to be input via env vars (which is sort of the preferred AWS/CI approach IIUC...

I wouldn't object that, but current "design" assumes that there can be several S3 endpoints, each with its own credentials, and one set of env vars cannot hold several key+secret pairs

Reply:
> Same for both - we don't want to write credentials to a local file

That's why those key+secret are generated on each run

Reply:
Suffering code is enterprise EAR, AWS KMS. It uses standard AWS env vars for credentials if not provided by scylla.yaml (also insecure) - same as aws SDK apps/cli tool etc. 

Typically, for CI, credentials like this are given via env, to, as @benipeled  says, avoid putting potentially confidential stuff in less secure files. Our Jenkins provides credentials for EAR testing (google and aws) as such.

What I would suggest is that in main.cc, the `convert<::object_storage_endpoint_param>` handler falls back to doing `std::getenv("AWS_...` iff the crediential info is _not_ in the config file. That way our CI can give AWS credentials in the preferred env way, but a deployment can give multiple (same as enterprise features).

Reply:
> What I would suggest is that in main.cc, the convert<::object_storage_endpoint_param> handler falls back to doing std::getenv("AWS_... iff the crediential info is not in the config file. That way our CI can give AWS credentials in the preferred env way, but a deployment can give multiple (same as enterprise features).

Good idea!